app/src/main/java/com/eveningoutpost/dexdrip/Models/Treatments.java

```diff
@@ -573,6 +573,14 @@ public static Treatments byTimestamp(long timestamp, int plus_minus_millis) {
                 .executeSingle();
     }
 
+    public static List<Treatments> listByTimestamp(long timestamp) {
```

Comment:
Do you really expect more than one treatment at a certain exact timestamp? Isn't `byTimestamp(long timestamp)` with its time range of 1.5 seconds and single return more appropriate, is it?

Reply:
Yes, because a bolus wizard will generate two entries at exactly the same time: carbs & insulin. Also an auto correction, notification or auto basal can occur any time.

Reply:
And you now always know the exact milliseconds of those events? If not, then I recommend to also use 1500 milliseconds near your expected time.

Reply:
I get the data from the CareLink system, which always returns exactly the same. Why should I extend the time range if I only want to know whether it is a new entry or an already existing one?

Reply:
Because it does not hurt and often there is a time shift of a few milliseconds.